src/Pact/Analyze/Analyze.hs

```diff
@@ -1368,221 +1122,176 @@ format s tms = do
+analyzePropSpecific (RowWriteCount tn pRk) = do
   sRk <- analyzeProp pRk
   tn' <- getLitTableName tn
   view $ qeAnalyzeState.rowWriteCount tn' sRk
 
 -- Authorization
-analyzeProp (KsNameAuthorized ksn) = nameAuthorized $ literalS ksn
-analyzeProp (RowEnforced tn cn pRk) = do
+analyzePropSpecific (KsNameAuthorized ksn) = nameAuthorized $ literalS ksn
+analyzePropSpecific (RowEnforced tn cn pRk) = do
   sRk <- analyzeProp pRk
   tn' <- getLitTableName tn
   cn' <- getLitColName cn
   view $ qeAnalyzeState.cellEnforced tn' cn' sRk
 
+
+-- TODO: move to Pact.Analyze.AnalyzePure
+analyzePure
+  :: (Analyzer m, SymWord a)
+  => PureTerm (ETermOf m) (TermOf m) a -> m (S a)
+analyzePure (Lit a)                    = pure (literalS a)
+analyzePure (Sym s)                    = pure s
+analyzePure (StrConcat p1 p2)          = (.++) <$> analyze p1 <*> analyze p2
+analyzePure (StrLength p)              = over s2Sbv SBV.length <$> analyze p
+analyzePure (Numerical a)              = analyzeNumerical a
+analyzePure (IntAddTime time secs)     = analyzeIntAddTime time secs
+analyzePure (DecAddTime time secs)     = analyzeDecAddTime time secs
+analyzePure (IntegerComparison op x y) = analyzeComparisonOp op x y
+analyzePure (DecimalComparison op x y) = analyzeComparisonOp op x y
+analyzePure (TimeComparison op x y)    = analyzeComparisonOp op x y
+analyzePure (StringComparison op x y)  = analyzeComparisonOp op x y
+analyzePure (BoolComparison op x y)    = analyzeComparisonOp op x y
+analyzePure (ObjectEqNeq op x y)       = analyzeObjectEqNeq  op x y
+analyzePure (KeySetEqNeq      op x y)  = analyzeEqNeq        op x y
+analyzePure (Logical op props)         = analyzeLogicalOp op props
+analyzePure (At schema colNameT objT retType)
+  = analyzeAt schema colNameT objT retType
+analyzePure LiteralObject {}
+  = error "literal object can't be an argument to analyzePure"
+analyzePure (Var vid name) = do
+  mVal <- getVar vid
+  case mVal of
+    Nothing                -> throwErrorNoLoc $ VarNotInScope name vid
+    Just (AVal mProv sval) -> pure $ mkS mProv sval
+    Just (AnObj obj)       -> throwErrorNoLoc $ AValUnexpectedlyObj obj
+    Just (OpaqueVal)       -> throwErrorNoLoc OpaqueValEncountered
+
+
+analyzePureO
+  :: Analyzer m
+  => PureTerm (ETermOf m) (TermOf m) Object -> m Object
+analyzePureO (LiteralObject obj) = Object <$> traverse analyzeE obj
+analyzePureO (At _schema colNameT objT _retType)
+  = analyzeAtO colNameT objT
+analyzePureO (Var vid name) = do
+  mVal <- getVar vid
+  case mVal of
+    Nothing            -> throwErrorNoLoc $ VarNotInScope name vid
+    Just (AVal _ val') -> throwErrorNoLoc $ AValUnexpectedlySVal val'
+    Just (AnObj obj)   -> pure obj
+    Just (OpaqueVal)   -> throwErrorNoLoc OpaqueValEncountered
+
+-- TODO(joel): I don't think an object can appear hear. Get more clarity on
+-- this.
+analyzePureO (Lit obj)     = pure obj
+analyzePureO (Sym _)       = vacuous "an object cannot be a symbolic value"
```

Comment:
We might want to consider our use of the word `vacuous` here if it's not too long before we [start using constraint vacuity checking](https://github.com/kadena-io/pact/blob/d76acad3e2b8e53711977abfbf5988a4a99bde15/src/Pact/Analyze/Check.hs#L400). This was my first thought when I saw the word.

Reply:
ah yeah. do you have a suggestion for a less ambiguous name? `vacuousCase`? `emptyPattern`? `vacuousMatch`?

Reply:
hm... perhaps `impossible`?